src/utilities/flattenRoutes.ts

```diff
@@ -1,31 +1,35 @@
 import { RouteFlat, Routes, isParentRoute, isNamedRoute } from '@/types'
+import { Path, combineParams, path as createPath } from '@/utilities'
 
-export function flattenRoutes(routes: Routes, path = ''): RouteFlat[] {
+export function flattenRoutes(routes: Routes, parentPath = '', parentParams = {}): RouteFlat[] {
   return routes.reduce<RouteFlat[]>((value, route) => {
-    const fullPath = path + route.path
+    const { params }: Path = typeof route.path === 'string' ? createPath(route.path, {}) : route.path
```

Comment:
fwiw I liked the utility you had for this before. Also, why alias the `path` utility?

Reply:
what utility? `path` at some point was reused, but now this is fixed

Reply:
You had a function called getParams or something that had this ternary and just returned the params.